### PR TITLE
Feature/patro23 123 add database integration testes

### PR DESCRIPTION
**Pull request by KlaudiaGajocha**
Database Integration Testes

### CODE REVIEW

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/DatabaseFixtures.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
+using DotNet.Testcontainers.Builders;
```
Comment (mateuszmuszynski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/DatabaseFixtures.cs:1]: Move classes to separate files

**Review by mateuszmuszynski — DISMISSED** [review]

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/MySQLTests.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using IContainer = DotNet.Testcontainers.Containers.IContainer;
+using System.Threading.Tasks;
+
+namespace Intive.Patronage2023.Modules.Example.Application.IntegrationTests;
+
+public class MsSqlTests : IAsyncLifetime
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/MySQLTests.cs:18]: The file has not the same name as class.

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		};
+		this._webApplicationFactory = new CustomWebApplicationFactory(fixture);
+	}
+
+	[Fact]
+	public async Task Handle_WhenCalled_ShouldReturnPagedList()
+	{
+		// Arrange
+		var scope = this._webApplicationFactory.Services.CreateScope();
+		var dbContext = scope.ServiceProvider.GetService<BudgetDbContext>();
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:34]: You can use GetRequiredService here?

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/MySQLTests.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		.WithImage("mcr.microsoft.com/mssql/server:2022-latest")
+		.WithPortBinding(MappedPort.ToString(), MsSqlPort.ToString())
+		.WithEnvironment("ACCEPT_EULA", "Y")
+		.WithEnvironment("SQLCMDUSER", Username)
+		.WithEnvironment("SQLCMDPASSWORD", Password)
+		.WithEnvironment("MSSQL_SA_PASSWORD", Password)
+		.WithWaitStrategy(Wait.ForUnixContainer().UntilCommandIsCompleted("/opt/mssql-tools/bin/sqlcmd", "-Q", "SELECT 1;"))
+		.Build();
+
+	public Task InitializeAsync()
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/MySQLTests.cs:36]: missing summary tag

File: src/api/app/Program.partial.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+namespace Intive.Patronage2023.Api;
+
+/// <summary>
+/// Docs.
```
Comment (mateuszgiedzinski) [inline src/api/app/Program.partial.cs:4]: invalid description of Program class.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs (hunk 3 above)
Comment (KlaudiaGajocha) [reply tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:34]: Good clue, thanks!

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+using Xunit.Sdk;
+
+namespace Intive.Patronage2023.Modules.Example.Application.IntegrationTests;
+
```
Comment (DawidBski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:19]: missing summary tag

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+		this.webApplicationFactory.Dispose();
+	}
+	///<summary>
+	///Custom web application factory used in the test to configure the test database connection string.
+	///</summary>
+	private class CustomWebApplicationFactory : WebApplicationFactory<Intive.Patronage2023.Api.Program>
+	{
+		private readonly string connectionString;
+
+		public CustomWebApplicationFactory(MsSqlTests fixture)
```
Comment (DawidBski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:101]: missing summary tag

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/MySQLTests.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+/// Integration tests for the MS SQL database using Testcontainers.
+/// </summary>
+public class MsSqlTests : IAsyncLifetime
+{
+	public const string Database = "patronage2023";
+	public const string Username = "sa";
+	public const string Password = "yourStrong(!)Password";
+	public const ushort MsSqlPort = 1433;
+	public const ushort MappedPort = 5000;
+
```
Comment (DawidBski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/MySQLTests.cs:27]: Maybe a short description here.

**Review by DawidBski — APPROVED** [review]
Some descriptions still missing but you've got my approve.

File: src/api/app/appsettings.json [hunk 9]
```diff
@@ -19,4 +19,5 @@
 		"InstrumentationKey": "",
 		"ConnectionString": ""
 	}
+
```
Comment (mateuszgiedzinski) [inline src/api/app/appsettings.json:22]: line break not necessary

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+namespace Intive.Patronage2023.Modules.Example.Application.IntegrationTests;
+
+public class IntegrationTests : IClassFixture<MsSqlTests>, IDisposable
+{
+	private readonly WebApplicationFactory<Intive.Patronage2023.Api.Program> webApplicationFactory;
+	/// <summary>
+	/// Initializes a new instance of the <see cref="IntegrationTests"/> class.
+	/// </summary>
+	/// <param name="fixture">The database fixture.</param>
+	///
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:27]: empty line not needed.

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+	private readonly WebApplicationFactory<Intive.Patronage2023.Api.Program> webApplicationFactory;
+	/// <summary>
+	/// Initializes a new instance of the <see cref="IntegrationTests"/> class.
+	/// </summary>
+	/// <param name="fixture">The database fixture.</param>
+	/// 
+	public IntegrationTests(MsSqlTests fixture)
+	{
+		this.webApplicationFactory = new CustomWebApplicationFactory(fixture);
+	}
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:31]: Empty line needed.

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+	///<summary>
+	///Unit test to verify that the GetBudgetsQueryHandler returns a PagedList of budget items.
+	///The test creates a budget item in the database, retrieves it using the query handler, and verifies that the result is not null and contains the expected item.
+	///</summary>
+	///
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:36]: empty line not needed.

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+		var handler = new GetBudgetsQueryHandler(dbContext);
+
+		// Act
+		var result = await handler.Handle(query, CancellationToken.None);
+
+		// Assert
+		result.Should().NotBeNull();
+		result.Items.Should().HaveCount(1);
+	}
+	///<summary>
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:87]: Empty line needed.

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+	///Disposes the web application factory used in the test.
+	///</summary>
+	public void Dispose()
+	{
+		this.webApplicationFactory.Dispose();
+	}
+	///<summary>
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:94]: empty line needed after method declaration.

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+using Xunit.Sdk;
+
+namespace Intive.Patronage2023.Modules.Example.Application.IntegrationTests;
+
+public class IntegrationTests : IClassFixture<MsSqlTests>, IDisposable
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/ExampleTests.cs:20]: Class not match file name.

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/MySQLTests.cs [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+using IContainer = DotNet.Testcontainers.Containers.IContainer;
+using System.Threading.Tasks;
+
+namespace Intive.Patronage2023.Modules.Example.Application.IntegrationTests;
+/// <summary>
+/// Integration tests for the MS SQL database using Testcontainers.
+/// </summary>
+public class MsSqlTests : IAsyncLifetime
+{
+	public const string Database = "patronage2023";
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/MySQLTests.cs:22]: missing summary tag for public visible consts.

File: app.props [hunk 17]
```diff
@@ -1,13 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 	<PropertyGroup>
-
-		<!-- Enable raising errors due not clean source code -->
```
Comment (mateuszgiedzinski) [inline app.props:5]: Of course these should not be deleted and all fixes should be done if any occurs.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/IntegrationTests.cs [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+using Microsoft.Extensions.DependencyInjection;
+
+using Xunit.Sdk;
+
+namespace Intive.Patronage2023.Modules.Example.Application.IntegrationTests;
+
+///<summary>
+///This class contains integration tests for the Example module of the Patronage2023 application.
+///</summary>
+
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/IntegrationTests.cs:23]: empty line not needed

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/IntegrationTests.cs [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+public class IntegrationTests : IClassFixture<MsSqlTests>, IDisposable
+{
+	private readonly WebApplicationFactory<Intive.Patronage2023.Api.Program> webApplicationFactory;
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="IntegrationTests"/> class.
+	/// </summary>
+	/// <param name="fixture">The database fixture.</param>
+
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/IntegrationTests.cs:32]: empty line not needed after summary tag. Please check all tags.

File: tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/Intive.Patronage2023.Modules.Example.Application.IntegrationTests.csproj [hunk 20]
```diff
@@ -0,0 +1,33 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
```
Comment (mateuszgiedzinski) [inline tests/integration/Intive.Patronage2023.Modules.Example.Application.IntegrationTests/Intive.Patronage2023.Modules.Example.Application.IntegrationTests.csproj:2]: app.props Import missing as it is in other projects.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: Intive.Patronage2023.sln [hunk 21]
```diff
@@ -53,6 +53,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Intive.Patronage2023.Module
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Intive.Patronage2023.Budget.Domain.Tests", "tests\unit\Intive.Patronage2023.Budget.Domain.Tests\Intive.Patronage2023.Budget.Domain.Tests.csproj", "{FC82E26F-63D3-47F9-80D1-098190A05D67}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "integration", "integration", "{505B0B82-9C7C-4A23-8BA3-A4F49BC2C0AB}"
```
Comment (mateuszgiedzinski) [inline Intive.Patronage2023.sln:56]: Please merge dev and resolve conflict.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

**Review by mateuszgiedzinski — APPROVED** [review]

**Pull request merged**